### PR TITLE
PD-8530 Deleting images throws an error if the number of image ids is…

### PR DESCRIPTION
**Pull request by mar4uk**
… 0 or multiples 20

### CODE REVIEW

File: admin_delete_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := s.bulkDeleteUploadedWithInvalidation(&mockHttpClient{}, ImageType, tc.publicIDs)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+type mockHttpClient struct {}
```
Comment (kirill-a-belov) [inline admin_delete_test.go]: Please use mock.Mock as we did all around the project.

File: admin_delete.go [hunk 2]
```diff
@@ -16,7 +16,11 @@ const (
 	pathUploadedRaws   = "/resources/raw/upload"
 )
 
-func (s *Service) bulkDeleteUploadedWithInvalidation(resourceType ResourceType, publicIDs []string) error {
+type HttpClient interface {
+	Do (req *http.Request) (*http.Response, error)
+}
+
+func (s *Service) bulkDeleteUploadedWithInvalidation(client HttpClient, resourceType ResourceType, publicIDs []string) error {
```
Comment (kirill-a-belov) [inline admin_delete.go]: I suggest putting the client on a service level and initializing it while servicing initialization.

**Review by kirill-a-belov — CHANGES_REQUESTED** [review]

File: admin_delete.go (hunk 2 above)
Comment (mar4uk) [reply admin_delete.go]: thanks! fixed

File: admin_delete_test.go (hunk 1 above)
Comment (mar4uk) [reply admin_delete_test.go]: fixed

Comment (mar4uk) [reply admin_delete_test.go]: replaced mock by httptest.
So changes in the method itself are minimal (no client dependency is needed anymore).

**Review by greenpart — APPROVED** [review]

**Pull request merged**